### ISSUE TITLE
macOS packaged Mu fails at running tkinter / turtle

### ISSUE THREAD

**Issue by tmontes** [mu-editor/pup#122]
See https://github.com/indygreg/python-build-standalone/issues/68

**Comment by tmontes** [comment]
CONFIRMED WORKAROUND:

* Setting the `TCL_LIBRARY` environment variable in macOS' app bundle template "executable script".

For example, hand-editing packaged Mu's `Mu Editor.app/Contents/MacOS/Mu Editor` script from...

```
#!/bin/sh

exec "$(dirname "$0")/../Resources/Python/bin/python3.7" -m mu "$@"
```

to:

```
#!/bin/sh

export TCL_LIBRARY="$(dirname "$0")/../Resources/Python/lib/tcl8.6/"
exec "$(dirname "$0")/../Resources/Python/bin/python3.7" -m mu "$@"
```

Makes `turtle` work from Mu's REPL. :-)

**Comment by tmontes** [comment]
UPDATE:

* The above mentioned `TCL_LIBRARY` environment variable and its need is now documented in the [Python Build Standalone docs](https://python-build-standalone.readthedocs.io/en/latest/quirks.html#tcl-tk-support-files) -- good.
* I left an outstanding thought/question to the author -- wondering if there could be a way to avoid setting the environment variable (one can hope, right?). :-)

HOWEVER:

* The likely near-term solution will be setting the `TCL_LIBRARY` environment variable and living with that.

**Comment by tmontes** [comment]
(self-note)

What looked like a quick fix -- grabbing the path to the PBS Tcl library from the `PYTHON.json` meta-data file, and using it in the macOS app bundle template -- actually requires a little bit more of work. Why:

* The current process is 1. instantiate the macOS bundle template, 2. extract the PBS distribution into it, 3. make adjustments/deletions/moves based on the PBS' `PYTHON.json` file, etc.
* This obviously won't work becase step 1. needs info from step 2./3.

Thus, a probable new approach, might be 1. extract the PBS distribution somewhere, 2. clean it up based on `PYTHON.json`, 3. instantiate the macOS app bundle template now knowing where the Tcl library will end up, 4. move the cleaned up PBS extracted distribution into the instantiated app bundle, etc.

*sigh...* :-)

**Comment by tmontes** [comment]
Keep this on hold until #128 is addressed.

**Comment by tmontes** [comment]
Closing after having merged #133.
